src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -363,8 +361,7 @@ private static boolean containsAllSafeTokens(final DetailAST expr) {
      */
     private static DetailAST skipVariableAssign(final DetailAST currentAST) {
         DetailAST result = currentAST;
-        if (currentAST.getType() == TokenTypes.ASSIGN
-                && currentAST.getFirstChild().getType() == TokenTypes.IDENT) {
+        if (currentAST.getType() == TokenTypes.ASSIGN) {
```

Comment:
Unsure how to kill this.

Reply:
An `ASSIGN` expressions requires a variable to store the value. So the first children will be an `IDENT` for sure.